ospfd/ospf_vty.c

```diff
@@ -8732,154 +9308,174 @@ static int config_write_ospf_distance(struct vty *vty, struct ospf *ospf)
-			}
-		for (ALL_LIST_ELEMENTS_RO(ospf->oiflist, node, oi)) {
-			if (!OSPF_IF_PARAM_CONFIGURED(oi->params,
-						      passive_interface))
-				continue;
-			if (OSPF_IF_PARAM_CONFIGURED(IF_DEF_PARAMS(oi->ifp),
-						     passive_interface)) {
-				if (oi->params->passive_interface
-				    == IF_DEF_PARAMS(oi->ifp)
-					       ->passive_interface)
-					continue;
-			} else if (oi->params->passive_interface
-				   == ospf->passive_interface_default)
-				continue;
+	/* RFC1583 compatibility flag print -- Compatible with CISCO
+	 * 12.1. */
+	if (CHECK_FLAG(ospf->config, OSPF_RFC1583_COMPATIBLE))
+		vty_out(vty, " compatible rfc1583\n");
 
-			vty_out(vty, " %spassive-interface %s %s\n",
-				oi->params->passive_interface ? "" : "no ",
-				oi->ifp->name,
-				inet_ntoa(oi->address->u.prefix4));
-		}
+	/* auto-cost reference-bandwidth configuration.  */
+	if (ospf->ref_bandwidth != OSPF_DEFAULT_REF_BANDWIDTH) {
+		vty_out(vty,
+			"! Important: ensure reference bandwidth "
+			"is consistent across all routers\n");
+		vty_out(vty, " auto-cost reference-bandwidth %d\n",
+			ospf->ref_bandwidth);
+	}
+
+	/* SPF timers print. */
+	if (ospf->spf_delay != OSPF_SPF_DELAY_DEFAULT
+	    || ospf->spf_holdtime != OSPF_SPF_HOLDTIME_DEFAULT
+	    || ospf->spf_max_holdtime != OSPF_SPF_MAX_HOLDTIME_DEFAULT)
+		vty_out(vty, " timers throttle spf %d %d %d\n",
+			ospf->spf_delay, ospf->spf_holdtime,
+			ospf->spf_max_holdtime);
 
-		/* Network area print. */
-		config_write_network_area(vty, ospf);
+	/* LSA timers print. */
+	if (ospf->min_ls_interval != OSPF_MIN_LS_INTERVAL)
+		vty_out(vty, " timers throttle lsa all %d\n",
+			ospf->min_ls_interval);
+	if (ospf->min_ls_arrival != OSPF_MIN_LS_ARRIVAL)
+		vty_out(vty, " timers lsa min-arrival %d\n",
+			ospf->min_ls_arrival);
 
-		/* Area config print. */
-		config_write_ospf_area(vty, ospf);
+	/* Write multiplier print. */
+	if (ospf->write_oi_count != OSPF_WRITE_INTERFACE_COUNT_DEFAULT)
+		vty_out(vty, " ospf write-multiplier %d\n",
+			ospf->write_oi_count);
 
-		/* static neighbor print. */
-		config_write_ospf_nbr_nbma(vty, ospf);
+	/* Max-metric router-lsa print */
+	config_write_stub_router(vty, ospf);
 
-		/* Virtual-Link print. */
-		config_write_virtual_link(vty, ospf);
+	/* SPF refresh parameters print. */
+	if (ospf->lsa_refresh_interval
+	    != OSPF_LSA_REFRESH_INTERVAL_DEFAULT)
+		vty_out(vty, " refresh timer %d\n",
+			ospf->lsa_refresh_interval);
 
-		/* Default metric configuration.  */
-		config_write_ospf_default_metric(vty, ospf);
+	/* Redistribute information print. */
+	config_write_ospf_redistribute(vty, ospf);
 
-		/* Distribute-list and default-information print. */
-		config_write_ospf_distribute(vty, ospf);
+	/* passive-interface print. */
+	if (ospf->passive_interface_default == OSPF_IF_PASSIVE)
+		vty_out(vty, " passive-interface default\n");
 
-		/* Distance configuration. */
-		config_write_ospf_distance(vty, ospf);
+	for (ALL_LIST_ELEMENTS_RO(om->iflist, node, ifp))
```

Comment:
`vrf_iflist(ospf->vrf_id)`?

The `om->iflist` list should be removed in my understanding. It's also being used in the `ospf_network_run()` function.